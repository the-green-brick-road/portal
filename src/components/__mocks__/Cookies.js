/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
