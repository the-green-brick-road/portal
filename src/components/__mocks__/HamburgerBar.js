/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
