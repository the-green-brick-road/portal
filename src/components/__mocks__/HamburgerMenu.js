/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;