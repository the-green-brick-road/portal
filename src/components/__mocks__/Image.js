/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
