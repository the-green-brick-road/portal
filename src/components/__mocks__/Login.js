/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
