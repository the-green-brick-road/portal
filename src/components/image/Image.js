/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
