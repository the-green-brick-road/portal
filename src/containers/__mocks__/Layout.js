/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment }   from 'react';
import { Outlet }     from "react-router-dom";

/* Material UI includes */
import { Typography } from '@mui/material';

function Layout(props) {

    return (

        <Fragment>
            <Typography> Mock Layout </Typography>
            <Outlet />
        </Fragment>

    );

}

export default Layout;
