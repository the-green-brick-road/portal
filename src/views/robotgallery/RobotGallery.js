/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# RobotGallery page definition
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment }   from 'react';

/* Material UI includes */
import { Typography } from '@mui/material';

function RobotGallery() {

    /* ----------- Define HTML --------- */
    return (
        <Fragment>
            <Typography> Robot gallery </Typography>
        </Fragment>
    );

}

export default RobotGallery;