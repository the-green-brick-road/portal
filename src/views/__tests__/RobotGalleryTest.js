/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Robot gallery view component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM }       from '@testing-library/react'
import { expect, test}             from '@jest/globals';

/* Component under test */
import { default as RobotGallery } from '../../views/robotgallery/RobotGallery';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { default as MockImage }    from '../../components/__mocks__/Image';
jest.mock("../../components", () => ({ Image: (props) => MockImage(props) }));
/* eslint-enable jest/no-mocks-import */

describe("RobotGallery view" ,() => {

    test('Should display robot gallery page', async () => {

        const view = render(
            <div>
                <RobotGallery/>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

})
