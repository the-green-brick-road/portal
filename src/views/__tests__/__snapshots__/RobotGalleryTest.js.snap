// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RobotGallery view Should display robot gallery page 1`] = `
"<body>
  <div>
    <div>
      <p
        class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
      >
         Robot gallery 
      </p>
    </div>
  </div>
</body>"
`;
